### PR TITLE
Detect merged branches. Affects #28.

### PR DESCRIPTION
**Pull request by robertdfrench**
This ensures we can test for "merged branches", which will subsequently be deleted as part of `pr-clean`.

### CODE REVIEW

**Comment by robertdfrench** [comment]
@calebwherry this PR uses the same testing patterns as #24, so if you don't like those, you might not like these either. I'm starting to have some concerns myself that `fake_git` does not serve any real purpose.

File: src/lib.rs [hunk 1]
```diff
@@ -221,4 +243,35 @@ mod tests {
         assert_eq!(pr_names[2], "dabba/doo/third");
         assert_eq!(pr_names[3], "fourth");
     }
+
+    #[test]
+    fn can_detect_merged_branches() {
+        let path = String::from("./target/release/fake_git");
```
Comment (calebwherry) [inline src/lib.rs]: Needs to be updated with new functionality on trunk.

File: src/lib.rs [hunk 2]
```diff
@@ -221,4 +243,35 @@ mod tests {
...
+        let fake_git = Git::with_path(path);
+        let merged_branches = fake_git.merged_branches().unwrap();
+        assert!(merged_branches.contains("already-been-merged"));
+    }
+
+    #[test]
+    fn can_issue_delete_statement() {
+        let path = String::from("./target/release/fake_git");
```
Comment (calebwherry) [inline src/lib.rs]: SAA

File: src/lib.rs [hunk 3]
```diff
@@ -221,4 +243,35 @@ mod tests {
...
+
+    #[test]
+    fn identify_branches_for_deletion() {
+        let mut merged_branches = vec![
+            "  one",
+            "* two",
+            "  trunk",
+            "  three"
+        ].join("\n");
+        merged_branches.push_str("\n");
```
Comment (calebwherry) [inline src/lib.rs]: Why can't this be the last element in the vec?

File: tests/real_git.rs [hunk 4]
```diff
@@ -75,3 +79,21 @@ fn can_list_all_branches() {
...
 }
+
+// Cleaning PRs requires that we identify "old" branches (those which have been merged into trunk),
+// and that we delete those branches. Because the tests run in parallel, we need to ensure that our
+// check for the existence of the "hotfix" branch always happens *before* our attempt to delete the
+// "hotfix" branch. So this test case exercises all the git client functionality we would need in
+// order to implement the "pr-clean" subcommand.
+#[test]
+fn could_clean() {
+    println!("TempDir='{:?}'", TEST_STATE.path());
```
Comment (calebwherry) [inline tests/real_git.rs:90]: Would be nice to push this into a macro I think to hide "touching" of tempdir. Maybe "init_test_state"? Or something like that.

Comment (calebwherry) [inline tests/real_git.rs:90]: Would be nice to push this into a macro I think to hide "touching" of tempdir. Maybe "init_test_state"? Or something like that.

**Review by calebwherry — CHANGES_REQUESTED** [review]
Reviewed.

File: tests/real_git.rs (hunk 4 above)
Comment (calebwherry) [reply tests/real_git.rs:90]: Created #40 to address this.

File: src/lib.rs (hunk 3 above)
Comment (calebwherry) [reply src/lib.rs]: Fixed in 63af7c6

File: src/lib.rs (hunk 2 above)
Comment (calebwherry) [reply src/lib.rs]: Fixed in 3ce3a48

File: src/lib.rs (hunk 1 above)
Comment (calebwherry) [reply src/lib.rs]: Fixed in 3ce3a48

**Review by calebwherry — APPROVED** [review]

**Pull request merged**